### PR TITLE
Fixes to track design manager

### PR DESCRIPTION
**Pull request by janisozaur**
Fix errorneous sizeof() argument, skip unneeded memset

### CODE REVIEW

File: src/ride/track_design.c [hunk 1]
```diff
@@ -123,9 +123,6 @@ static rct_track_td6 *track_design_open_from_buffer(uint8 *src, size_t srcLength
 	rct_track_td6 *td6 = calloc(1, sizeof(rct_track_td6));
 	uint8 *readPtr = src;
 
-	// Clear top of track_design as this is not loaded from the td4 files
-	memset(&td6->track_spine_colour, 0, 67);
```
Comment (janisozaur) [inline src/ride/track_design.c:127]: This `memset` has wrong argument: tries to set 67 bytes of a 4-byte item and is not needed, as the space for object has been `calloc`ed.

File: src/ride/track_design_save.c [hunk 2]
```diff
@@ -472,7 +472,7 @@ static void track_design_save_pop_map_element_desc(rct_object_entry *entry, int
 			memmove(
 				&_trackSavedMapElementsDesc[removeIndex],
 				&_trackSavedMapElementsDesc[removeIndex + 1],
-				remainingNumItems * sizeof(rct_td6_scenery_element*)
+				remainingNumItems * sizeof(rct_td6_scenery_element)
```
Comment (janisozaur) [inline src/ride/track_design_save.c:475]: It looks like this is what you intended to do.

File: src/ride/track_design.c (hunk 1 above)
Comment (IntelOrca) [inline src/ride/track_design.c]: Are you sure? It should be dereferencing the field on the struct.

Comment (IntelOrca) [inline src/ride/track_design.c:127]: Its not wrong, its setting several fields - not just that one. But as you say, it should already by zeroed by calloc.

**Pull request merged**